application/backend/app/services/training/subset_assignment/distribution.py

```diff
@@ -0,0 +1,54 @@
+# Copyright (C) 2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from app.models import DatasetItemSubset
+
+from .models import SplitRatios
+
+
+class SubsetDistribution:
+    """Represents current distribution of items across subsets."""
+
+    def __init__(self, counts: dict[DatasetItemSubset, int]):
+        self._counts = counts.copy()
+
+    @property
+    def total(self) -> int:
+        return sum(self._counts.values())
+
+    def get_count(self, subset: DatasetItemSubset) -> int:
+        return self._counts.get(subset, 0)
+
+    def compute_adjusted_ratios(self, target_ratios: SplitRatios, unassigned_count: int) -> SplitRatios:
+        """
+        Compute adjusted ratios for unassigned items to reach target distribution.
+
+        This method calculates how unassigned items should be distributed to achieve the overall target ratios.
+        """
+        if unassigned_count == 0:
+            return target_ratios
```

Comment:
```suggestion
        if unassigned_count < 0:
            raise ValueError(f"Parameter 'unassigned_count' cannot be negative (got {unassigned_count})")

        if unassigned_count == 0:
            return target_ratios
```
Just some extra validation, for safety